packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -4691,6 +4696,72 @@ class _UpdateTextSelectionToAdjacentLineAction<T extends DirectionalCaretMovemen
   bool get isActionEnabled => state._value.selection.isValid;
 }
 
+class _UpdateTextSelectionToAdjacentPageAction<T extends DirectionalCaretMovementIntent> extends ContextAction<T> {
+  _UpdateTextSelectionToAdjacentPageAction(this.state);
+
+  final EditableTextState state;
+
+  VerticalCaretMovementRun? _verticalMovementRun;
```

Comment:
Is your vertical run logic still necessary on Mac for shift + pageup/down?  If so what's the behavior that it produces?  I thought I had that covered in my PR but now I'm doubting myself.

Reply:
Yes, it's necessary whenever the selection is updated so that the new extent, say, 20 lines below, has the right horizontal offset too, depending on where the run started.
It would be the equivalent of pressing (Shift+) arrowDown multiple times.

Reply:
Sounds good.  I guess https://github.com/flutter/flutter/pull/105497 will wait for this PR to be merged first.

FYI you have some analyzer errors on this PR by the way.